intermediate_source/reinforcement_q_learning.py

```diff
@@ -30,30 +30,24 @@
 
 The CartPole task is designed so that the inputs to the agent are 4 real
 values representing the environment state (position, velocity, etc.).
-However, neural networks can solve the task purely by looking at the
-scene, so we'll use a patch of the screen centered on the cart as an
-input. Because of this, our results aren't directly comparable to the
-ones from the official leaderboard - our task is much harder.
-Unfortunately this does slow down the training, because we have to
-render all the frames.
+We take these 4 inputs without any scaling and pass them through a 
+small fully-connected network with 2 outputs, one for each action. 
+The network is trained to predict the expected value for each action, 
+given the input state. The action with the highest expected value is 
+then chosen.
 
-Strictly speaking, we will present the state as the difference between
-the current screen patch and the previous one. This will allow the agent
-to take the velocity of the pole into account from one image.
 
 **Packages**
 
 
 First, let's import needed packages. Firstly, we need
 `gym <https://github.com/openai/gym>`__ for the environment
-(Install using `pip install gym`).
+(Install using `pip install gym`). Developed on v0.26.1 of gym.
```

Comment:
Can we please remove: `Developed on v0.26.1 of gym.` We are updating version of gym to 0.25 which should be sufficient to run this tutorial.

Reply:
I would have thought that this only works w/ v0.26 and above because of the truncation vs termination logic. Did this actually run on v0.25? This is mentioned in the `Breaking backward compatibility` section of https://github.com/openai/gym/releases/tag/0.26.0

Reply:
@malfet any thoughts?

Reply:
Why do we want to make this tutorial exclusive to latest gym version? I.e. I was able to run the tutorial in its current form using both gym-0.25 and gym-0.26, what is wrong with keeping it backward compatible with older version (which by the way is currently preinstalled in Google Colab)

Is there a way to test if ipyb generated from this tutorial is runnable in Colab without any weaks (which imo is an important feature of tutorials, isn't it?)

Reply:
I originally added this note due to a comment from @pseudo-rnd-thoughts. I don't have a strong opinion, but I understood that v0.26 had major and breaking changes, so we may want to steer users away from those older gym versions. @pseudo-rnd-thoughts might be able to elaborate further as one of the maintainers of gym.

Reply:
Hi, Gym v26 implemented a number of breaking API changes that while v25 has implemented, by default are turned off. Therefore, to run this tutorial, I believe you are going to have to modify several parameters in `gym.make`, `env.reset`, etc. 
Whatever version of the tutorial we need to make it clear what Gym version we are developing for due to the breaking changes noted above.

Reply:
Anybody knows of the plans of https://github.com/Kautenja/nes-py to be compatible with gym v26? Also, any hint when Google Colab can change its default version from v25 to v26?

Reply:
Sure, but v0.26 still supports compatible mode, so one can tweak the tutorial pretty easily, i.e. something like that should work:
```
env = gym_super_mario_bros.make("SuperMarioBros-1-1-v0", render_mode='rgb', apply_api_compatibility=True)
```

Reply:
https://github.com/pytorch/tutorials/pull/2075

Reply:
An initial look at the tutorial that solution would only work for v0.25 and v0.26 so it would be simpler to just have a single version supported. In addition, at a quick look I think the new step API implementation is done incorrectly. I will see if I can update it soon